### PR TITLE
fix(audible): stop misclassifying fiction as nonfiction

### PR DESCRIPTION
**Pull request by djdembeck**
## Summary

Classify `literatureType` from Audible `category_ladders` data instead of searching `thesaurus_subject_keywords` for the word "fiction".

## Motivation

The previous keyword-based approach checked whether any `thesaurus_subject_keywords` entry contained the string `"fiction"`. This was brittle: keywords are free-text and inconsistently populated, so fiction classification could be wrong for books that lacked the literal word "fiction" in their keyword list.

Audible's `category_ladders` provides structured browse-node hierarchy — the top-level category is a far more reliable signal. Each region has known fiction root category IDs; walking the ladder to find a matching root gives accurate fiction/nonfiction classification.

## Changes

- **New module** `src/helpers/books/audible/literatureType.ts` (345 lines): region-specific fiction root category ID tables for US, UK, DE, FR, IT, ES, JP, CA, AU, plus the `literatureTypeFromProduct()` classifier that walks category ladders to find a fiction root match
- **ApiHelper**: replaced inline keyword check with call to `literatureTypeFromProduct()`, reading `category_ladders`, `thesaurus_subject_keywords`, and `region`
- **Keyword fallback preserved**: if category ladders are absent or empty, falls back to the original keyword-based heuristic
- **Test dataset**: `B0036I54I6` expected `literatureType` corrected from `nonfiction` to `fiction` (the keyword-based check had it wrong)
- **New test file**: `tests/helpers/books/audible/literatureType.test.ts` covers fiction/nonfiction classification across regions and empty-missing ladder edge cases

## Notes

- Keyword fallback is intentionally retained as a safety net for edge cases where category ladders are missing or empty
- The classifier must run before `getGenres()` since that method mutates `category_ladders`
- Region tables are manually curated from known Audible browse node IDs; adding new regions requires updating the table

## Summary by CodeRabbit

* **Improvements**
  * Improved Audible book classification using category information, subject keywords, and regional rules.
  * Added support for localized literature categories across multiple markets, including Germany and Japan.
  * Improved handling of fiction, nonfiction, children’s, and teen book categories.

* **Bug Fixes**
  * Corrected titles that were previously misidentified as nonfiction.

* **Tests**
  * Added coverage for regional categories, keyword fallback, and missing classification data.

### CODE REVIEW

File: src/helpers/books/audible/literatureType.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+			'Romance',
+			'Science Fiction & Fantasy'
+		],
+		mixedChildren: {
+			"Children's Audiobooks": [
+				'Action & Adventure',
+				'Fairy Tales, Folk Tales & Myths',
+				'Humour',
+				'Literature & Fiction',
+				'Mystery & Suspsense',
```
Comment (nimuebot) [inline src/helpers/books/audible/literatureType.ts:148]: **should-fix** — copy-paste typo in the IN table: `'Mystery & Suspsense'` is missing the second `p`. The US list on the same table uses `'Mystery & Suspense'`, so `children.includes(ladder[1]?.name)` will never match the real IN browse node. Result: an IN `Children's Audiobooks > Mystery & Suspense` book is classified `nonfiction` instead of `fiction`. Fix the spelling.

File: src/helpers/books/audible/literatureType.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+	region: string
+): LiteratureType {
+	const table = LITERATURE_TABLES[region] ?? LITERATURE_TABLES['us']
+	if (categoryLadders.length > 0) {
+		const matches = categoryLadders.some(({ ladder }) => {
+			const root = ladder[0]?.name
+			if (!root) return false
+			if (table.fictionRoots.includes(root)) return true
+			const children = table.mixedChildren[root]
+			return !!children && children.includes(ladder[1]?.name ?? '')
```
Comment (nimuebot) [inline src/helpers/books/audible/literatureType.ts:337]: **should-fix/question** — the `mixedChildren` strings drift from the `fictionRoots` strings within the same table, so books under mixed roots often won't match. Examples: FR `'Comédie et humour': ['Littérature et fiction']` and `Jeunesse: ['Roman et littérature']` vs the FR root `'Littérature, romans et fiction'`; ES `LGBTQ+` lists `'Misterio, negra y suspense'` vs the ES root `'Policíaca, negra y suspense'`. Since matching here is exact-string `includes()` (line 337), a real child node spelled like the canonical root silently flips that book to `nonfiction`. These tables are hand-curated with no live verification. Consider validating them against real browse data — the repo already has a live-test workflow (`tests/live/`, `RUN_LIVE_TESTS`) built exactly for catching [AUDIBLE HTML CHANGE] drift — or at minimum a per-region test that asserts every `mixedChildren` value appears somewhere in the region's ladder vocabulary.

File: src/helpers/books/audible/literatureType.ts [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+	}
+}
+
+export default function literatureTypeFromProduct(
+	categoryLadders: AudibleProduct['product']['category_ladders'],
+	thesaurusKeywords: string[] | undefined,
+	region: string
+): LiteratureType {
+	const table = LITERATURE_TABLES[region] ?? LITERATURE_TABLES['us']
+	if (categoryLadders.length > 0) {
```
Comment (nimuebot) [inline src/helpers/books/audible/literatureType.ts:331]: **nit** — the PR body advertises a fallback for "absent or empty" ladders, but this guard only handles empty (`length > 0`). An *absent* `category_ladders` field would throw here before the keyword fallback runs. It's safe today because `AudibleProductSchema` declares `category_ladders` as required (`z.array(...)`, types.ts:171) and every ladder-less response still yields `[]` upstream, but the repo just hardened against omitted Audible fields (e.g. #905). A `categoryLadders ?? []` at this call site (or the ApiHelper caller) would make the advertised fallback actually reachable and match that precedent for nearly free.

File: src/helpers/books/audible/ApiHelper.ts [hunk 4]
```diff
@@ -288,6 +289,12 @@ class ApiHelper {
 	 */
 	getFinalData(): ApiBook {
 		if (!this.audibleResponse) throw new Error(ErrorMessageNoData(this.asin, 'ApiHelper'))
+		// Classify literature type BEFORE getGenres() which mutates category_ladders
+		const literatureType = literatureTypeFromProduct(
```
Comment (nimuebot) [inline src/helpers/books/audible/ApiHelper.ts:293]: **heads-up/question for the maintainer** — because `category_ladders` is in the always-requested param set (ApiHelper.ts:60), effectively every product now classifies strictly from the curated tables, and the keyword fallback only fires for ladder-less responses. That means `literatureType` flips for a broad set of books, not just the demonstrably-misclassified ones (a biographical/historical work under a root not in `fictionRoots` that previously matched a `fiction` keyword now becomes `nonfiction`). This is the intended tradeoff per the PR, but it's a default-behaviour change across the whole catalog — worth an explicit maintainer call. Consider whether a definitive `fiction` keyword should outrank an unknown/ambiguous root rather than being ignored whenever any ladder exists.

File: tests/datasets/audible/books/api.ts [hunk 5]
```diff
@@ -62,11 +63,11 @@ export function setupMinimalParsed(
 		isbn: response.isbn ?? '',
 		isAdult: response.is_adult_product,
 		language: response.language,
-		literatureType: response.thesaurus_subject_keywords?.some((keyword) =>
-			keyword.includes('fiction')
-		)
-			? 'fiction'
-			: 'nonfiction',
+		literatureType: literatureTypeFromProduct(
```
Comment (nimuebot) [inline tests/datasets/audible/books/api.ts:66]: **nit** — `setupMinimalParsed` hardcodes `'us'` as the region while classifying `response.category_ladders`, so any future non-US fixture gets US tables. Also note both sides of the dataset-vs-ApiHelper assertion now call the same `literatureTypeFromProduct`, so that comparison is largely tautological for this field — the explicit `fiction` literal on `minimalB0036I54I6` (line 1293) and the new unit tests carry the real assertion weight. No action required; flagging so a future region-specific fixture doesn't silently misclassify.

**Review by nimuebot — COMMENTED** [review]
Ranked **P2** (fix). The approach is sound and tightly scoped: replacing the keyword-contains `fiction` check with a category-ladder root traversal fixes a real misclassification (B0036I54I6) and the call is correctly ordered before `getGenres()` mutates the shared ladder arrays. New unit tests pass; the 58 failures in `tests/helpers/books/audible/` are pre-existing (identical on `origin/develop`, prototype-binding issues in `ApiHelper.test.ts`, unrelated to this diff). All 10 region keys in `#static/regions` are covered by `LITERATURE_TABLES`.

Five findings, two substantive:
- **should-fix** `literatureType.ts:148` — typo `'Mystery & Suspsense'` in the IN table silently wrong-classifies IN children's mystery books as `nonfiction`.
- **should-fix** `literatureType.ts:337` — `mixedChildren` strings drift from `fictionRoots`/real browse-node names (FR, ES); exact-string `includes()` matching means real books silently flip to `nonfiction`. Worth validating the curated tables against live browse data (the repo has `tests/live/` for exactly this).

Nit / heads-up: the keyword fallback is effectively dead in the normal path since `category_ladders` is always requested — `literatureType` now shifts for the whole catalog, a default-behaviour change that needs your explicit sign-off (also flagged inline); minor points on absent-vs-empty ladder handling, the harness's hardcoded `'us'`, and the harness/SUT assertion being partly tautological.

None of these are correct to block on — they're touch-ups plus a maintainer call. Nice, well-documented fix — thanks for the clear PR.

File: src/helpers/books/audible/literatureType.ts [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+		],
+		mixedChildren: {
+			"Children's Audiobooks": [
+				'Action & Adventure',
+				'Fairy Tales, Folk Tales & Myths',
+				'Humour',
+				'Literature & Fiction',
+				'Mystery & Suspsense',
+				'Science Fiction & Fantasy'
+			],
```
[the remark this answers is not in the document]
Comment (djdembeck) [reply src/helpers/books/audible/literatureType.ts:150]: Our internal review flagged this, this is actually a misspelling from audible themselves

File: src/helpers/books/audible/literatureType.ts (hunk 1 above)
Comment (djdembeck) [reply src/helpers/books/audible/literatureType.ts:148]: Our internal review flagged this, this is actually a misspelling from audible themselves

File: src/helpers/books/audible/literatureType.ts [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+			'コメディー・落語': ['ユーモア・風刺文学・フィクション'],
+			ティーン: [
+				'SF・ファンタジー',
+				'文学・フィクション・ライトノベル',
+				'ミステリー・スリラー・サスペンス',
+				'ロマンス'
+			],
+			'絵本・児童書': [
+				'SF・ファンタジー',
```
[the remark this answers is not in the document]
Comment (djdembeck) [reply src/helpers/books/audible/literatureType.ts:314]: Acknowledged — valid finding. Fixing in this branch.

File: tests/helpers/books/audible/literatureType.test.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+		expect(literatureTypeFromProduct(ladders, [], 'us')).toBe('fiction')
+	})
+
+	test('comedy biography (2nd-level nonfiction) stays nonfiction', () => {
+		const ladders: Ladders = [cat(['Comedy & Humor', 'Biographies & Memoirs'])]
+		expect(literatureTypeFromProduct(ladders, [], 'us')).toBe('nonfiction')
+	})
+
+	test('keyword fallback when no ladders', () => {
+		const ladders: Ladders = []
```
[the remark this answers is not in the document]
Comment (djdembeck) [reply tests/helpers/books/audible/literatureType.test.ts:47]: Acknowledged — valid finding. Fixing in this branch.
